### PR TITLE
 Add 3 new uses cases (add home-buyer, add homes and view home-seller based on priority), update non-functional requirements and glossary

### PR DESCRIPTION
**Pull request by KhoonSun47**
The developer guide lacks information about new use cases. Hence, adding new use cases ensures that developers have comprehensive documentation about the system's functionality.

The new use cases include adding home-buyer, adding homes and viewing home-seller based on priority.

Non-functional requirements and glossary are also added, so that the developers have a comprehensive understanding of the system's performance, security, usability, and other operational characteristics, in addition to a clear reference for technical terms used throughout the documentation.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -385,6 +388,145 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 3j. EstateEase detects missing role in the entered data. <br>
+    * 3j1. EstateEase shows an error message regarding missing role. <br>
+      Use case resumes from step 2.
+
+* 3k. EstateEase detects incorrect role in the entered data. <br>
+    * 3k1. EstateEase shows an error message regarding the entry of incorrect role. <br>
+      Use case resumes from step 2.
+
+**Use case: UC00 - Add more homes to home-seller**
```
Comment (lokidoki102) [inline docs/DeveloperGuide.md:450]: 😂I think renumber all the UC instead. First one should be UC01 not UC00.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -385,6 +388,145 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 3j. EstateEase detects missing postal code in the entered data. <br>
+    * 3j1. EstateEase shows an error message regarding missing postal code. <br>
+      Use case resumes from step 2.
+
+* 3k. EstateEase detects incorrect format for postal code in the entered data. <br>
+    * 3k1. EstateEase shows an error message regarding incorrect format for postal code. <br>
+      Use case resumes from step 2.
+
+**Use case: UC00 - Viewing of homes based on home-seller's priority**
```
Comment (lokidoki102) [inline docs/DeveloperGuide.md:515]: Change the UC# here too

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -385,6 +388,145 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
     * 3q1. EstateEase shows an error message regarding the entry of incorrect priority level. <br>
       Use case resumes from step 2.
 
+**Use case: UC00 - Add home-buyer to user’s contact list**
```
Comment (lokidoki102) [inline docs/DeveloperGuide.md:391]: This one too.

**Review by lokidoki102 — CHANGES_REQUESTED** [review]
Content is okay, just change the numbering in ascending order from top to bottom, starting from UC01 (i.e. UC01, UC02, ... UC##).

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (KhoonSun47) [reply docs/DeveloperGuide.md]: Yeap updated the numbering of the use case. Thanks for reminding.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (KhoonSun47) [reply docs/DeveloperGuide.md]: Yeap updated the numbering of the use case. Thanks for reminding.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (KhoonSun47) [reply docs/DeveloperGuide.md]: Yeap updated the numbering of the use case. Thanks for reminding.

**Review by lokidoki102 — APPROVED** [review]
LGTM

**Review by redcolorbicycle — APPROVED** [review]
Khoon Sun? King Sun.

**Pull request merged**